src/Umbraco.Core/Persistence/Repositories/ITrackedReferencesRepository.cs

```diff
@@ -70,6 +70,32 @@ IEnumerable<RelationItemModel> GetPagedRelationsForItem(
         bool filterMustBeIsDependency,
         out long totalRecords);
 
+    /// <summary>
+    ///     Gets a paged result of items which are in relation with an item in the recycle bin.
+    /// </summary>
+    /// <param name="objectTypeKey">The Umbraco object type that has recycle bin support (currently Document or Media).</param>
+    /// <param name="skip">The amount of items to skip.</param>
+    /// <param name="take">The amount of items to take.</param>
+    /// <param name="filterMustBeIsDependency">
+    ///     A boolean indicating whether to filter only the RelationTypes which are
+    ///     dependencies (isDependency field is set to true).
+    /// </param>
+    /// <param name="totalRecords">The total count of the items with reference to the current item.</param>
+    /// <returns>An enumerable list of <see cref="RelationItem" /> objects.</returns>
+    IEnumerable<RelationItemModel> GetPagedRelationsForRecycleBin(
+        Guid objectTypeKey,
+        long skip,
+        long take,
+        bool filterMustBeIsDependency,
+        out long totalRecords)
+    {
+        totalRecords = 0;
+        return [];
+    }
+
+    Task<PagedModel<RelationItemModel>> GetPagedRelationsForRecycleBinAsync(UmbracoObjectTypes objectType, long skip, long take, bool filterMustBeIsDependency)
+        => Task.FromResult(new PagedModel<RelationItemModel>(0, []));
```

Comment:
Maybe I'm missing something, but we are we adding this if it isn't used anywhere, and not implemented? Just for future proofing? If so maybe it should wrap the non-async, so it does something :smile:

Reply:
Sorry for confusion - you've just found an aborted look at moving paging up to the repository level.  It seemed too tricky so I took another route and then didn't clean this up.  I'll do so.

Reply:
No worries :smile: Makes sense why it looked out of place then :smile: